design-system/src/theme-provider.tsx

```diff
@@ -48,7 +48,7 @@ const applyTheme = ({
   const vars = toVars(
     themeOverrides && isObject(themeOverrides)
       ? merge({}, themes.default, themes[validTheme], themeOverrides)
-      : themes[validTheme]
+      : merge({}, themes.default, themes[validTheme])
```

Comment:
I believe here we need to merge the new theme with the default one, otherwise only the new theme tokens are used.